### PR TITLE
ref: Use deletions code path for cleanup script.

### PR DESCRIPTION
**Pull request by bretthoerner**
- [x] Implement concurrency for deletions code path.

### CODE REVIEW

File: src/sentry/deletions/base.py [hunk 1]
```diff
@@ -131,7 +132,10 @@ def chunk(self):
         query_limit = self.query_limit
         remaining = self.chunk_size
         while remaining > 0:
-            queryset = list(self.model.objects.filter(**self.query)[:query_limit])
+            queryset = self.model.objects.filter(**self.query)
+            if self.order_by:
+                queryset = queryset.order_by(self.order_by)
+            queryset = list(queryset[:query_limit])
```
Comment (bretthoerner) [inline src/sentry/deletions/base.py]: I implemented the `order_by` thing, but didn't even up using it.

I think the `EventTag` ordering doesn't matter since we are using the deletion code child relation stuff to delete them by ID: 

https://github.com/getsentry/sentry/blob/8330a0938a896d813a342be1e1ca2237da4a6074/src/sentry/deletions/defaults/event.py#L29-L31

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/runner/commands/cleanup.py [hunk 2]
```diff
@@ -147,77 +146,76 @@ def is_filtered(model):
                     project=project or '*',
                 )
             )
+
+        # TODO: filtering won't take into account "app-level" cascades done in deletions code...
+        #       e.g. filtering out GroupTagValue won't work when we delete a Group here, because all
+        #            child relations are handled within the deletions code.
```
Comment (bretthoerner) [inline src/sentry/runner/commands/cleanup.py]: Not sure if we care or not? If we do, it means I'd have to push down filtering of models to the deletion code.

File: src/sentry/runner/commands/cleanup.py [hunk 3]
```diff
@@ -147,77 +146,76 @@ def is_filtered(model):
...
-    if not silent:
-        click.echo("Cleaning up unused FileBlob references")
-    if is_filtered(models.FileBlob):
-        if not silent:
-            click.echo('>> Skipping FileBlob')
-    else:
-        cleanup_unused_files(silent)
+            has_more = True
+            while has_more:
+                has_more = task.chunk()
```
Comment (bretthoerner) [inline src/sentry/runner/commands/cleanup.py]: As discussed, I'm doing this without a work queue for now. I do want to add the concurrency back in though.

File: src/sentry/runner/commands/cleanup.py [hunk 4]
```diff
@@ -147,77 +146,76 @@ def is_filtered(model):
...
+
+    # Deletions that use `BulkDeleteQuery` (and don't need to worry about child relations)
+    # (model, datetime_field, max_days)
+    BULK_QUERY_DELETES = (
+        (models.TagValue, 'last_seen', None),
+        (models.EventMapping, 'date_added', 7),
+    )
```
Comment (bretthoerner) [inline src/sentry/runner/commands/cleanup.py]: You might be asking yourself, "why aren't these just in the tuple above, to be handled by the deletions code path?"

The problem is that they both use `BulkModelDeletionTask` in the deletions code:

https://github.com/getsentry/sentry/blob/8330a0938a896d813a342be1e1ca2237da4a6074/src/sentry/deletions/__init__.py#L73

https://github.com/getsentry/sentry/blob/8330a0938a896d813a342be1e1ca2237da4a6074/src/sentry/deletions/__init__.py#L45

`ModelDeletionTask` passes the `query` down to the Django ORM:

https://github.com/getsentry/sentry/blob/8330a0938a896d813a342be1e1ca2237da4a6074/src/sentry/deletions/base.py#L134

But `BulkModelDeletionTask` inherits from `ModelDeletionTask` but then uses the `query` in a hardcoded `WHERE x = y`:

https://github.com/getsentry/sentry/blob/96bc1c63df5ec73fe12c136ada11561bf52f1ec9/src/sentry/utils/query.py#L292-L293

In short, it doesn't currently support doing `date_field < value`. Since they don't have child relations anyway, I've left them on the (faster) `BulkDeleteQuery` path for now.

**Comment by bretthoerner** [comment]
@mattrobenolt I need to add more tests and work on the currency thing but I'd appreciate any WTFs you have here when you have the time.

File: src/sentry/runner/commands/cleanup.py (hunk 2 above)
Comment (mattrobenolt) [reply src/sentry/runner/commands/cleanup.py]: I don't think this is worth considering. In theory, it'd already be deleted, or will be deleted real soon anyways.

**Review by mattrobenolt — APPROVED** [review]
This all seems chill to me. If you want to add more tests, I won't stop you. Probably a good idea to do so.

cc @JTCunning

**Comment by bretthoerner** [comment]
👍 Ya I want to try some tests because delete cost as some of my first platform works scares the tf out of me.
